ClrMD.Extensions/ClrDynamic.cs

```diff
@@ -130,8 +135,11 @@ public bool IsUndefined()
 
         private static ClrInstanceField FindField(ObfuscatedField oField)
         {
-            TypeName delcaringType = ClrMDSession.Current.ObfuscateType(oField.DeclaringType);
-            var target = ClrMDSession.Current.Heap.GetTypeByName(delcaringType);
+            TypeName declaringType = ClrMDSession.Current.ObfuscateType(oField.DeclaringType);
+
+            var heapTypes = ClrMDSession.Current.Heap.EnumerateTypes();
+
+            var target = heapTypes.First(type => type.Name == declaringType);
```

Comment:
This will iterate all types to find the declaringType everytime ClrDynamic.FindField is called. This probably introduces a performance bottleneck.

If there isn't a way to lookup by type name, a cached map should be preprocessed to do this only once